Source/JavaScriptCore/inspector/protocol/Page.json

```diff
@@ -24,6 +24,27 @@
                 "WebSecurityEnabled"
             ]
         },
+        {
+            "id": "UserPreference",
+            "type": "object",
+            "description": "A user preference that can be overriden by Web Inspector, like an accessibility preference.",
+            "properties": [
+                { "name": "name", "$ref": "UserPreferenceName", "description": "Preference name." },
+                { "name": "value", "$ref": "UserPreferenceValue", "description": "Preference value." }
+            ]
+        },
+        {
+            "id": "UserPreferenceName",
+            "type": "string",
+            "enum": ["PrefersReducedMotion", "PrefersContrast"],
+            "description": "User preference name."
+        },
+        {
+            "id": "UserPreferenceValue",
+            "type": "string",
+            "enum": ["Off", "On"],
```

Comment:
I thought the plan was to have this be a `"type": "any"` since each user presence could be something that could have different values (e.g. `prefers-contrast: more` vs `prefers-reduced-motion: reduce` vs etc.).

Reply:
I don't understand how `"type": "any"` makes it any different.

For the upcoming patch that implements overriding color scheme (appearance), the protocol types are augmented like this:


```diff
        {
             "id": "UserPreferenceName",
             "type": "string",
-            "enum": ["PrefersReducedMotion", "PrefersContrast"],
+            "enum": ["PrefersReducedMotion", "PrefersContrast", "PrefersColorScheme"],
             "description": "User preference name."
         },
                 {
            "id": "UserPreferenceValue",
            "type": "string",
-            "enum": ["Off", "On"],
+            "enum": ["Off", "On", "Light", "Dark"],
            "description": "User preference value."
        },
```

The implementation of `InspectorPageAgent::overrideUserPreference(Protocol::Page::UserPreferenceName, std::optional<Protocol::Page::UserPreferenceValue>&&)` explicitly checks the preference name against expected values to set the overrides, ignoring the unexpected.

With `"type": "any"` wouldn't we do the same on the backend?

Reply:
i suppose yeah it would be nice to have a way for the frontend to check for value support (in addition to the name support you're adding here)

in that case, we should change `"On"` and `"Off"` to be the actual values used by the relevant name(s) (e.g. `"no-preference"`, `"reduce"`, `"less"`, `"more"`, etc.)

Reply:
Ok. I added explicit preferences values: "NoPreference", "More" and "Reduce". 
They're handled on the backend corresponding to their preference name.